### PR TITLE
boards: olimex stm32-h407: Add initial support

### PR DESCRIPTION
**Pull request by rettichschnidi**
Olimex STM32-E407 is very similar and got used as template.

Signed-off-by: Reto Schneider <code@reto-schneider.ch>

### CODE REVIEW

File: boards/arm/olimex_stm32_h407/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Kconfig - OLIMEX-STM32-H407 board configuration
+#
+# Copyright (c) 2017, Erwin Rol <erwin@erwinrol.com>
```
Comment (ydamigos) [inline boards/arm/olimex_stm32_h407/Kconfig.board]: You introduce a new file you don't need to keep this copyright. Please remove it.

File: boards/arm/olimex_stm32_h407/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Kconfig - OLIMEX-STM32-H407 board configuration
+#
+# Copyright (c) 2017, Erwin Rol <erwin@erwinrol.com>
```
Comment (ydamigos) [inline boards/arm/olimex_stm32_h407/Kconfig.defconfig]: Same here

File: boards/arm/olimex_stm32_h407/board.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2015 Intel Corporation
```
Comment (ydamigos) [inline boards/arm/olimex_stm32_h407/board.h]: Add your copyright.

File: boards/arm/olimex_stm32_h407/olimex_stm32_h407.dts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2017, Erwin Rol <erwin@erwinrol.com>
```
Comment (ydamigos) [inline boards/arm/olimex_stm32_h407/olimex_stm32_h407.dts]: Please remove the old copyright.

File: boards/arm/olimex_stm32_h407/pinmux.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
```
Comment (ydamigos) [inline boards/arm/olimex_stm32_h407/pinmux.c]: Please add your copyright.

**Review by ydamigos — CHANGES_REQUESTED** [review]
Looks good. Some comments regarding copyright.

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks

**Comment by rettichschnidi** [comment]
@ydamigos But it feels a little mean to remove their copyright. They have done 99% of the work. Anyway, I just did the requested changes and push-forced them.

**Review by ydamigos — APPROVED** [review]
@rettichschnidi LGTM. Regarding copyright, I just expressed my opinion. Other people may comment on it.

**Comment by erwango** [comment]
> It seems a little bit unfair to me to remove their copyright altogether. After all, they have done 99% of the work.

I agree with @ydamigos.
While I agree it sounds right from you to  copy the initial copyright of the file you copied, this method does not upscale and can't be applied systematically and hence practically:
For board ports, everyone copy a port from another board and this is applied from the beginning. So theoretically, if reporting the initial copyright was applied every time there would be a long list of copyrights in each file (or only 2 if everyone pick up systematically the same example, which I doubt).
Besides, this would even not represent the exact impact of each contributor on the file. Indeed, in time, code evolution implies that all files are modified along with generic changes (done by other contributors) and current files might look quite different from the initial version committed by the initial contributor.
This is why it is preferred to retain the copyright of the one who made the effort to port, test and document the board (which is you) rather than the initial person that also ported another board in (a bit different way) some time back.

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**